src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -91,10 +99,25 @@ public static InputModelType CreateModelType(ref Utf8JsonReader reader, string?
                 }
             }
 
-            return model ?? CreateInputModelTypeInstance(id, name, ns, accessibility, deprecated, description, usageString, discriminatorValue, discriminatorPropertyName, baseModel, properties, inheritedDictionaryType, isNullable, resolver);
+            return model ?? CreateInputModelTypeInstance(id, name, ns, accessibility, deprecated, description, usageString, discriminatorValue, discriminatorPropertyName, baseModel, properties, inheritedDictionaryType, isNullable, mediaTypes, resolver);
         }
 
-        private static InputModelType CreateInputModelTypeInstance(string? id, string? name, string? ns, string? accessibility, string? deprecated, string? description, string? usageString, string? discriminatorValue, string? discriminatorPropertyValue, InputModelType? baseModel, IReadOnlyList<InputModelProperty> properties, InputDictionaryType? inheritedDictionaryType, bool isNullable, ReferenceResolver resolver)
+        private static void CreatMediaTypes(ref Utf8JsonReader reader, ICollection<string> mediaTypes, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartArray)
+            {
+                throw new JsonException();
+            }
+            reader.Read();
+
+            while (reader.TokenType != JsonTokenType.EndArray)
+            {
+                var type = reader.ReadWithConverter<string>(options);
+                mediaTypes.Add(type ?? throw new JsonException($"null item isn't allowed"));
+            }
+            reader.Read();
+        }
```

Comment:
well this is a list of string, I think we do not have to write this manually, just call
```
reader.TryReadWithConverter()
```
and give it a `IReadOnlyList<string>`, it will work.

Reply:
I will try